src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py

```diff
@@ -663,5 +664,48 @@ def on_goto_request(self, py_db, request):
         response = pydevd_base_schema.build_response(request, kwargs={'body': {}})
         return NetCommand(CMD_RETURN, 0, response, is_json=True)
 
+    def _can_set_dont_trace_pattern(self, py_db, start_patterns, end_patterns):
+        if py_db.is_cache_file_type_empty():
+            if py_db.dont_trace_external_files.__name__ == 'dont_trace_files':
+                return py_db.dont_trace_external_files.start_patterns == start_patterns and \
+                    py_db.dont_trace_external_files.end_patterns == end_patterns
+            else:
+                # File type cache is empty and don't trace is not set.
+                return True
+        return False
+
+    def on_setdebuggerproperty_request(self, py_db, request):
+        args = request.arguments.kwargs
+        if 'dontTraceStartPatterns' in args and 'dontTraceEndPatterns' in args:
+            start_patterns = tuple(args['dontTraceStartPatterns'])
+            end_patterns = tuple(args['dontTraceEndPatterns'])
+            if self._can_set_dont_trace_pattern(py_db, start_patterns, end_patterns):
+                def dont_trace_files(abs_path):
```

Comment:
I think that this method could be renamed to `dont_trace_files_property_request`.

-- that way when checking `if py_db.dont_trace_external_files.__name__ == 'dont_trace_files'` I think it helps a bit in understanding that it's a patched function.